include/pybind11/detail/native_enum_data.h

```diff
@@ -29,10 +29,12 @@ class native_enum_data {
     native_enum_data(const object &parent_scope,
                      const char *enum_name,
                      const char *native_type_name,
+                     const char *enum_doc,
```

Comment:
You're using "class doc" in the PR title and the added test: I think it's a much better name than `enum_doc` here, because then it is crystal clear that it does not apply to the enum members. Could you please use `class_doc` here (and in native_enum.h), too?